src/Chainweb/Version/Guards.hs

```diff
@@ -60,14 +60,29 @@ import Chainweb.Transaction
 import Chainweb.Version
 import Chainweb.Utils.Rule
 
-getForkHeight :: Fork -> ChainwebVersion -> ChainId -> BlockHeight
+getForkHeight :: Fork -> ChainwebVersion -> ChainId -> ForkHeight
 getForkHeight fork v cid = v ^?! versionForks . at fork . _Just . onChain cid
 
 checkFork
-    :: (BlockHeight -> BlockHeight -> Bool)
+    :: (BlockHeight -> ForkHeight -> Bool)
     -> Fork -> ChainwebVersion -> ChainId -> BlockHeight -> Bool
 checkFork p f v cid h = p h (getForkHeight f v cid)
 
+after :: BlockHeight -> ForkHeight -> Bool
+after bh (ForkAtBlock bh') = bh > bh'
+after _ ForkAtGenesis = True
+after _ ForkNever = False
+
+atOrAfter :: BlockHeight -> ForkHeight -> Bool
+atOrAfter bh (ForkAtBlock bh') = bh >= bh'
+atOrAfter _ ForkAtGenesis = True
+atOrAfter _ ForkNever = False
+
+before :: BlockHeight -> ForkHeight -> Bool
+before bh (ForkAtBlock bh') = bh < bh'
+before _ ForkAtGenesis = False
+before _ ForkNever = True
```

Comment:
I'm really tempted to delete `before` entirely and rename the forks that use it, so that we have less special behavior in here.